### PR TITLE
689 fix deep search function

### LINKED ISSUE #689: fix deepSearch function
There is currently a method called `deepSearch` that lives in `/lib/deepSearch.js`.
It is incomplete and needs to be fixed!

deepSearch searches through an object/array and return a list of all objects that match the test function

### Acceptance Criteria:
1. Running `yarn test deepSearch` results in tests passing.
2. You have written a skeleton method for someone else to work on.
3. You have written tests surrounding your skeleton method.
4. Running `yarn lint` does not print any errors to the console!
5. Optional: write one or two more tests surrounding the method you are solving to account for potential edge cases.

### Please include the skeleton of a new method + an accompanying test for someone else to work on at the time of creating a pull request! A pull request will most likely be denied if it does not contain a skeleton method for someone else to work on! For more information, please read the [Contributing Guide](https://github.com/BlakeGuilloud/ganon/blob/master/CONTRIBUTING.md).

Thank you so much for your contribution!

### PR DESCRIPTION
**Pull request by gracemgately**
Closes #689 

- [x] Running `yarn lint` does not trigger any linter errors
- [x] The test of the method you have fixed is passing
- [x] You have written a new skeleton method for someone else to work on!
- [x] You have written tests to accompany your skeleton method

### CODE REVIEW

File: test/deepSearch.test.js [hunk 1]
```diff
@@ -23,25 +23,22 @@ const testData = {
...
-        { type: "string", value: "test1" },
-        { type: "string", value: "test2" },
-        { type: "string", value: "test3" }
-      ])
-    );
+      deepSearch(testData, item => item.type === "string")).toEqual([
+      { type: "string", value: "test1" },
+      { type: "string", value: "test2" },
+      { type: "string", value: "test3" }
+    ]);
```
Comment (ktilcu) [inline test/deepSearch.test.js]: Looks like some formatting changes were introduced. I don't think they are right either. Cna you back these out?

Comment (gracemgately) [reply test/deepSearch.test.js]: Sure! The reason I put those changes in was because the tests didn't seem to be running correctly with the expect block parentheses ordered the first way -- it was looking for the results of the function to have 'toEqual' property?

Comment (gracemgately) [reply test/deepSearch.test.js]: i.e. this would pass all the 'deep search test' specs: 
```
  return {
    toEqual: () => []
  }
```

Comment (gracemgately) [reply test/deepSearch.test.js]: @ktilcu Given ^, would you still like me to take those changes out?

**Comment by gracemgately** [comment]
@BlakeGuilloud will do! thank you for letting me work on this :)
